homeassistant/components/netatmo/data_handler.py

```diff
@@ -0,0 +1,168 @@
+            for data_class in islice(self._queue, 0, PARALLEL_CALLS):
+                if data_class["next_scan"] > time():
+                    continue
+                self._data_classes[data_class["name"]]["next_scan"] = (
+                    time() + data_class["interval"]
+                )
+                try:
+                    self.data[
+                        data_class["name"]
+                    ] = await self.hass.async_add_executor_job(
+                        partial(data_class["class"], **data_class["kwargs"]),
+                        self._auth,
+                    )
+                    async_dispatcher_send(
+                        self.hass, f"netatmo-update-{data_class['name']}"
+                    )
+                except (pyatmo.NoDevice, pyatmo.ApiError) as err:
+                    _LOGGER.debug(err)
+
+            self._queue.rotate(PARALLEL_CALLS)
+
+        async_track_time_interval(
+            self.hass, async_update, timedelta(seconds=SCAN_INTERVAL)
+        )
+
+        self.listeners.append(
+            self.hass.bus.async_listen("netatmo_event", self.handle_event)
+        )
+
+    async def async_remove(self):
+        """Clean up the Netatmo data handler."""
+        for listener in self.listeners:
+            listener()
+
+    async def handle_event(self, event):
+        """Handle webhook events."""
+        if event.data["data"]["push_type"] == "webhook_activation":
+            _LOGGER.info("%s webhook successfully registered", MANUFACTURER)
+            self._webhook = True
+
+    async def register_data_class(self, data_class_name, **kwargs):
+        """Register data class."""
+        if "home_id" in kwargs:
+            data_class_entry = f"{data_class_name}-{kwargs['home_id']}"
+        elif "area_name" in kwargs:
+            data_class_entry = f"{data_class_name}-{kwargs.pop('area_name')}"
+        else:
+            data_class_entry = data_class_name
+
+        async with self.lock:
+            if data_class_entry not in self._data_classes:
+                self._data_classes[data_class_entry] = {
+                    "class": DATA_CLASSES[data_class_name],
+                    "name": data_class_entry,
+                    "interval": DEFAULT_INTERVALS[data_class_name],
+                    "next_scan": time() + DEFAULT_INTERVALS[data_class_name],
+                    "kwargs": kwargs,
+                    "registered": 1,
+                }
+
+                try:
+                    self.data[
+                        data_class_entry
+                    ] = await self.hass.async_add_executor_job(
+                        partial(DATA_CLASSES[data_class_name], **kwargs), self._auth,
+                    )
+                except (pyatmo.NoDevice, pyatmo.ApiError) as err:
+                    _LOGGER.debug(err)
+
+                self._queue.append(self._data_classes[data_class_entry])
+                _LOGGER.debug("Data class %s added", data_class_entry)
```

Comment:
This code is duplicated in async_update above.  Can we make it another function?